### PR TITLE
#112: update the squid API spec.

### PR DESCRIPTION
**Pull request by ssallam**

### CODE REVIEW

**Review by Pedro-vk — APPROVED** [review]
LGTM :+1:

File: doc/architecture/squid.md [hunk 1]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
...
+---
 
-ASYNC. Returns a list of orders by given account.
+#### consume
 
-```js
-array[Order] = ocean.getOrdersByAccount(account)
+Parameters
+```
+        agreementId:
```
Comment (aaitor) [inline doc/architecture/squid.md]: missing definition

File: doc/architecture/squid.md [hunk 2]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
...
 
-```js
-array[Order] = ocean.getOrdersByAccount(account)
+Parameters
+```
+        agreementId:  
+                did: str the asset did which consist of `did:op:` and the assetId hex str (without `0x` prefix)
+serviceDefinitionId: str id of the service within the asset DDO
+    consumerAccount: Account instance of the consumer ordering the service
+               mode: ?
```
Comment (aaitor) [inline doc/architecture/squid.md]: what is the mode parameter?

File: doc/architecture/squid.md [hunk 3]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
...
 
-ASYNC. Signs the ServiceAgreement from the consumer side
-```
-signature,serviceAgreementId = ocean.signServiceAgreement(did, serviceDefinitionId, consumer)
-```
+| Method              | Return Value                        | Python | JS   | Java |
+| :------------------ | :-----------------------------------| :----- | :--- |:---- |
+| list                | Account list                        |        |      |      |
+| balance             | tuple(ether_balance, ocean_balance) |        |      |      |
+| request_tokens      |                                     |        |      |      |
```
Comment (aaitor) [inline doc/architecture/squid.md]: missing return value

File: doc/architecture/squid.md [hunk 4]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
...
-id= account.getId()
+const amount = 30
+const account = ocean.accounts.list()[0]
+const success = ocean.accounts.requestTokens(account, amount)
 ```
 
-### getOceanBalance
+---
+
+## ocean.secret_store
```
Comment (aaitor) [inline doc/architecture/squid.md]: Are not private functions?
If are private I'm okay specifying here but detailing are private. 
I would add references to the existing Secret Store documentation in dev-ocea

**Review by aaitor — CHANGES_REQUESTED** [review]
Nice advance @ssallam , please take a look at the comments but this start to have a good shape

File: doc/architecture/squid.md [hunk 5]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
...
-/**
- * Extends OceanBase providing Account functionalities
-*/
-class Account extends OceanBase  {
-    name: string
-    balance: Balance
-}
+| Method                             | Return Value            | Python | JS   | Java |
+| :--------------------------------- | :---------------------- | :----- | :--- | :--- |
+| Ocean.getInstance() / Ocean()      | Ocean instance             |        |      |      |
```
[the remark this answers is not in the document]
Comment (ssallam) [reply doc/architecture/squid.md]: The table is a compact list of all functions in a module, and shows the status of implementation in the different languages.

We can move this to a separate page if does not seem to fit here.

File: doc/architecture/squid.md [hunk 6]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
 
 **Note 1: "Provider" was renamed to Aquarius and some of its functionality was moved over to Brizo. This page hasn't been updated to reflect that change yet.**
```
[the remark this answers is not in the document]
Comment (ssallam) [reply doc/architecture/squid.md]: Will add it as suggested.

File: doc/architecture/squid.md [hunk 7]
```diff
@@ -4,662 +4,709 @@ The Squid API is a Level-2 API built on top of core Ocean components. It's a fac
...
```
[the remark this answers is not in the document]
Comment (ssallam) [reply doc/architecture/squid.md]: Deleted.

File: doc/architecture/squid.md (hunk 1 above)
Comment (ssallam) [reply doc/architecture/squid.md]: Added the definition.

File: doc/architecture/squid.md (hunk 2 above)
Comment (ssallam) [reply doc/architecture/squid.md]: Replaced it for now with `resultPath`

**Review by aaitor — APPROVED** [review]

**Pull request merged**